usersession/userd/settings.go

```diff
@@ -127,17 +152,17 @@ func safeSnapFromSender(s *Settings, sender dbus.Sender) (string, *dbus.Error) {
 	return snap, nil
 }
 
-func desktopFileFromValue(s *Settings, command string, setting string, dotDesktopValue string, sender dbus.Sender) (string, *dbus.Error) {
+func desktopFileFromValue(s *Settings, command string, setspec *settingSpec, dotDesktopValue string, sender dbus.Sender) (string, *dbus.Error) {
 	snap, err := safeSnapFromSender(s, sender)
 	if err != nil {
 		return "", err
 	}
-	if err := settingWhitelisted(setting); err != nil {
+	if err := setspec.validate(); err != nil {
```

Comment:
Feels like the validate is not really related to converting dotDesktopValue to corresponding snap_app.desktop. Perhaps it bit should be done similarly to `Get()/GetSub()` in the caller.

Reply:
Maybe true, but a bit unrelated to the overall refactoring here. I personally don't have a strong opinion either way.

Reply:
if we rename it to the very long desktopFileFromValueForSetting is probably clearer and a reasonable factoring